### PR TITLE
Allow users to use custom metric name prefix for Stackdriver

### PR DESCRIPTION
**Pull request by songy23**
Fixes https://github.com/census-instrumentation/opencensus-java/issues/1350

Equivalent to https://github.com/census-instrumentation/opencensus-go/pull/583.

### CODE REVIEW

**Comment by songy23** [comment]
Holding off this change - we're going to discuss about this a little in tomorrow's meeting.

**Comment by songy23** [comment]
Closing for now - with our new Metrics exporting approach this should no longer be needed.

File: exporters/stats/stackdriver/src/test/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtilsTest.java [hunk 1]
```diff
@@ -185,6 +185,26 @@ public void createMetric() {
...
+  public void createMetric_WithExternalMetricDomain() {
+    View view =
+        View.create(
+            Name.create(VIEW_NAME),
+            VIEW_DESCRIPTION,
+            MEASURE_DOUBLE,
+            DISTRIBUTION,
+            Arrays.asList(KEY),
+            CUMULATIVE);
+    String prometheusDomain = "external.googleapis.com/prometheus/";
```
Comment (songy23) [inline exporters/stats/stackdriver/src/test/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtilsTest.java]: /cc @jkohen 

Does this match what you suggested in https://github.com/census-instrumentation/opencensus-go/issues/836?

Comment (jkohen) [reply exporters/stats/stackdriver/src/test/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtilsTest.java]: Yes, thank you.

File: exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java [hunk 2]
```diff
@@ -81,13 +81,15 @@
...
   @VisibleForTesting static final String OPENCENSUS_TASK_DESCRIPTION = "Opencensus task identifier";
+  @VisibleForTesting static final String DEFAULT_DISPLAY_NAME_PREFIX = "OpenCensus";
   private static final String GCP_GKE_CONTAINER = "gke_container";
   private static final String GCP_GCE_INSTANCE = "gce_instance";
   private static final String AWS_EC2_INSTANCE = "aws_ec2_instance";
   private static final String GLOBAL = "global";
 
   private static final Logger logger = Logger.getLogger(StackdriverExportUtils.class.getName());
   private static final String CUSTOM_METRIC_DOMAIN = "custom.googleapis.com";
+  private static final String EXTERNAL_METRIC_DOMAIN = "external.googleapis.com";
```
Comment (bogdandrutu) [inline exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java]: What is this? Where is this coming from?

File: exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java [hunk 3]
```diff
@@ -234,8 +238,32 @@ static MetricDescriptor createMetricDescriptor(View view, String projectId) {
     return builder.build();
   }
 
-  private static String generateType(String viewName) {
-    return CUSTOM_OPENCENSUS_DOMAIN + viewName;
+  private static String generateType(
+      String viewName, @javax.annotation.Nullable String metricNamePrefix) {
+    String domain;
+    if (metricNamePrefix == null) {
```
Comment (bogdandrutu) [inline exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java]: Can we cache this logic? I mean the calculation of the domain. Is this does not happen for all the points then we can just add a comment to explain that.

**Pull request merged**